doc/dbus_api.md

```diff
@@ -576,6 +606,43 @@ string `AccessType` could be replaced by a boolean `ReadOnly`).
 * `PartitionInfo`: Partition names (and sometimes their type) separated by commas.
   Eg. "_/dev/dasda1 (Linux native), /dev/dasda2 (Linux native)_". Empty if the information is unknown.
 
+#### `org.opensuse.DInstaller.Storage1.Job` Interface
+
+This interface is implemented by objects exported at the `/org/opensuse/DInstaller/Storage1/jobs/[0-9]+`
+paths. It provides information about a long-running process.
+
+##### Properties
+
+~~~
+Running  readable b
+ExitCode readable u
+~~~
+
+* `Running`: Whether the Job is still being executed or it has already finished.
+* `ExitCode`: Final result. Zero for running processes.
+
+##### Signals
+
+* `Finished(u exit_code)`: the Job is not longer running and the exit code has been set to its final value.
+
+#### `org.opensuse.DInstaller.Storage1.DASD.Format` Interface
+
+This interface is implemented by those Job objects used to represent the formatting of a set of DASDs.
+
+##### Properties
+
+~~~
+Summary readable a{s(uub)}
+~~~
+
+* `Summary`: A hash where each key is the path of one of the DASDs being formatted and the value is
+  the status represented by a triplet of total cylinders, cylinders already processed and a boolean
+  indicating whether the process for that particular disk is completed.
+
+##### Signals
+
+* `SummaryUpdated(a{s(uub)} progress)`: the summary has been updated with new progress information for some of the DASDs.
```

Comment:
This should not be necessary, the [standard signal `PropertiesChanged`](https://dbus.freedesktop.org/doc/dbus-specification.html#standard-interfaces-properties) is designed for it and implemented by ruby-dbus

Reply:
Well, it's a bit different here. The property "Summary" is a hash like:

```
{
  /path/to/dasds/1 => [2000, 100, false],
  /path/to/dasds/2 => [2500, 150, false],
  /path/to/dasds/3 => [2500, 200, false],
  /path/to/dasds/4 => [2000, 100, false],
  /path/to/dasds/5 => [3000, 120, false],
}
```

Typically there are more entries (maybe 20?).

Now assume that only the formatting of dasds 3 and 4 have progressed. As far as I understand `PropertiesChanged` the whole hash would be in that signal.

But with the current definition of `SummaryUpdated` only the two entries for the DASDs 3 and 4 are part of the signal. I don't even know if that's important/relevant or whether that small saving pays off. But that's the reason why we decided to have a specific signal instead of using `PropertiesChanged`.

Reply:
See the [detailed docs for `dbus_reader`](https://www.rubydoc.info/gems/ruby-dbus/0.19.0/DBus/Object#dbus_reader-class_method)

Reply:
Aah, I missed that subtle detail

Reply:
So
```suggestion
* `SummaryUpdated(a{s(uub)} progress)`: the summary has been updated with new progress information for **some** of the DASDs.
```

Reply:
Let me recap our options:

1. standard `PropertiesChanged` only: least surprising, but it may repeat too much data
2. `SummaryUpdated`: to be useful we should omit `PropertiesChanged`, or more precisely, omit its *value*, declaring it as `invalidates`.
3. @joseivanlopez proposed a `LastFormattedDevices` property using `PropertiesChanged`. IMHO that is more confusing than (2)

Reply:
Ok, sorry for the confusion on IRC. Now I got the problem after reading the code. So, we have:

* A `Summary` property which does not emit a `PropertiesChanged` signal when it value changes.
* A `SummaryUpdated` signal that is emitted when the `Summary` changes, but it only contains the modified parts of the summary.

Having properties without emitting a `PropertiesChanged` signal sounds a bit weird to me. Some clients like cockpitjs expect it, otherwise the proxy never udpates the value of the property. I would try to follow the same approach as the rest of D-Bus APIs in D-Installer, so I would emit a `PropertiesChanged` signal when Summary changes. And, if we do that, maybe we don't need the `SummaryUpdated` signal anymore. We could include another value for each entry in the summary to indicate if the value is new or not.

Reply:
> we should check that cockpit.dbus supports it.

Apparently it does: when it gets a signal that Summary was invalidated, it asks for it with Get.
https://github.com/cockpit-project/cockpit/blob/2a8a630ed0a56dc1c9c4946eae997d902b16383a/src/bridge/cockpitdbuscache.c#L969

Reply:
Changed at b3bab6998866c15dfa5c45, where `SummaryUpdated` is dropped in favor of the more standard `PropertiesChanged`.